OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -1362,6 +1413,31 @@ internal static void UpgradeWeaponRules(int engineVersion, ref List<MiniYamlNode
 					}
 				}
 
+				if (engineVersion < 20150517)
+				{
+					// Remove DeathType from DamageWarhead
+					if (node.Key.StartsWith("Warhead") && node.Value.Value == "SpreadDamage")
+					{
+						var deathTypeNode = node.Value.Nodes.FirstOrDefault(x => x.Key == "DeathType");
+						var deathType = deathTypeNode == null ? "1" : FieldLoader.GetValue<string>("DeathType", deathTypeNode.Value.Value);
+						var damageTypes = node.Value.Nodes.FirstOrDefault(x => x.Key == "DamageTypes");
+						if (damageTypes != null)
+							damageTypes.Value.Value += ", DeathType" + deathType;
+						else
+							node.Value.Nodes.Add(new MiniYamlNode("DamageTypes", "DeathType" + deathType));
+
+						node.Value.Nodes.RemoveAll(x => x.Key == "DeathType");
+					}
+
+					// Replace "DeathTypeX" damage types with proper words
+					if (node.Key.StartsWith("Warhead") && node.Value.Value == "SpreadDamage")
```

Comment:
They are actually different upgrade rules rules, doing different things with the same nodes.
Also the first will apply to any and all mods and maps, while the second only cares about the official mods (I suggest RA2 be updated manually to keep YAML quality up).